pkg/informer/informer.go

```diff
@@ -216,11 +217,15 @@ func (d *DynamicDiscoverySharedInformerFactory) AddIndexers(indexers cache.Index
 
 func (d *DynamicDiscoverySharedInformerFactory) Start(ctx context.Context) {
 	// Immediately discover types and start informing.
-	// TODO: Feed any failure to discover types into /readyz, instead of
-	// panicking.
-	if err := d.discoverTypes(ctx); err != nil {
-		// TODO(sttts): don't klog.Fatal, but return an error
-		klog.Fatalf("Error discovering initial types: %v", err)
+	if err := wait.PollImmediateInfiniteWithContext(ctx, time.Second, func(ctx context.Context) (bool, error) {
```

Comment:
should we tear down any informers when the ctx has been closed?

Reply:
we do. See the defer.

Reply:
defer will be executed only if `PollImmediateInfiniteWithContext` ends with success, so we might not clean up once the ctx has been closed and we had some errors.

Reply:
but nothing is started before PollImmediateInfiniteWithContext succeeds